### PR TITLE
Hw 04  http pt1

### PR DESCRIPTION
**Pull request by SteveRusin**

### CODE REVIEW

File: messages/message.go [hunk 1]
```diff
@@ -3,10 +3,13 @@ package messages
...
+	"sync"
 )
 
 type MessageDb map[string][]Message
 
+var mutex *sync.Mutex = &sync.Mutex{}
+
 type Message struct {
 	Id   string
 	Text string
```
Comment (sergii-rft) [inline messages/message.go:15]: It would be more beneficial to add mutex to the struct

File: messages/message.go [hunk 2]
```diff
@@ -3,10 +3,13 @@ package messages
 import (
 	"errors"
 	"fmt"
+	"sync"
 )
 
 type MessageDb map[string][]Message
 
+var mutex *sync.Mutex = &sync.Mutex{}
+
```
Comment (sergii-rft) [inline messages/message.go:12]: It would be more beneficial to add mutex to the struct в MessageRepository

File: users/user_handler.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		w.Write([]byte("Invalid username/password\n"))
+
+		return
+	}
+
+	response := LoginUserResponse{
+		Url: "ws://mock.url.io/token=.....",
+	}
+
+	json.NewEncoder(w).Encode(response)
```
Comment (sergii-rft) [inline users/user_handler.go:84]: No error check

File: users/user_handler.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+func HandleUserLogin(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		fmt.Fprintf(w, "404 Method not found\n")
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	loginUserDto := LoginUserDto{}
+	json.NewDecoder(r.Body).Decode(&loginUserDto)
```
Comment (sergii-rft) [inline users/user_handler.go:67]: No error check

File: users/user_handler.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+var usersRepository = NewUserRepository()
+
+func HandleUser(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		fmt.Fprintf(w, "404 Method not found\n")
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	registerUserDto := RegisterUserDto{}
+	json.NewDecoder(r.Body).Decode(&registerUserDto)
```
Comment (sergii-rft) [inline users/user_handler.go:36]: No error check
can return 500

File: users/user_handler.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+type LoginUserResponse struct {
+	Url string `json:"url"`
+}
+
+var usersRepository = NewUserRepository()
+
+func HandleUser(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		fmt.Fprintf(w, "404 Method not found\n")
```
Comment (sergii-rft) [inline users/user_handler.go:31]: don't have return statement, status code will be 200

File: users/user_handler.go [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+type LoginUserDto = RegisterUserDto
+
+type LoginUserResponse struct {
+	Url string `json:"url"`
+}
+
+var usersRepository = NewUserRepository()
+
+func HandleUser(w http.ResponseWriter, r *http.Request) {
```
Comment (sergii-rft) [inline users/user_handler.go:29]: You can include this method in the user's structure, allowing you to incorporate usersRepository as a field within it for use in the handler

Alternatively, you can create a closure to encapsulate and manage the usersRepository variable.

File: messages/message.go (hunk 2 above)
Comment (SteveRusin) [reply messages/message.go:12]: let's leave it as is for now. This will be replaced with database anyway

File: users/user_handler.go (hunk 7 above)
Comment (SteveRusin) [reply users/user_handler.go:29]: Thanks for the suggestion!
Let's leave it as is.

**Pull request merged**